### DISCUSSION TITLE
Need to dump the network logs captured by mitmproxy into a new file, without stopping or disrupting the running service.

### DISCUSSION THREAD

**Discussion by samyakLambda** [mitmproxy/mitmproxy#7181] [Q&A]
I need to capture network logs on a real Android or iOS device without interrupting the mitmproxy service that is running in the background. Currently, the way it works is that I only get the log file after shutting down the mitmproxy service(the logs are stored by mitm in its buffer until shutDown). However, what I need is the ability to extract or dump the logs multiple times while mitmproxy is still running. In other words, I want to capture the logs at different intervals without having to stop and restart the mitmproxy process each time.

**Comment by emanuele-em** [comment]
My advice is to use addons:
with addons you can write a little python script that write flows on a file, and integrate it with mitmproxy is easy.
[Here](https://docs.mitmproxy.org/stable/addons-examples/#io-write-flow-file) you find a skeleton code, it could be a good starting point.

To run mitmproxy with your script just run `mitmdump -s ./yourscript.py`

[Addons documentation](https://docs.mitmproxy.org/stable/addons-overview/)
